specs/lightclient/validator.md

```diff
@@ -0,0 +1,198 @@
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Prerequisites](#prerequisites)
+- [Custom types](#custom-types)
+- [Containers](#containers)
+  - [SyncCommitteeContribution](#synccommitteecontribution)
+- [Validator assignments](#validator-assignments)
+  - [Lookahead](#lookahead)
+- [Beacon chain responsibilities](#beacon-chain-responsibilities)
+  - [Block proposal](#block-proposal)
+    - [Preparing a `BeaconBlock`](#preparing-a-beaconblock)
+    - [Constructing the `BeaconBlockBody`](#constructing-the-beaconblockbody)
+      - [Sync committee](#sync-committee)
+    - [Packaging into a `SignedBeaconBlock`](#packaging-into-a-signedbeaconblock)
+  - [Attesting and attestation aggregation](#attesting-and-attestation-aggregation)
+  - [Sync committee contributions](#sync-committee-contributions)
+  - [Sync committee contribution aggregation](#sync-committee-contribution-aggregation)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This document represents the expected behavior of an "honest validator" with respect to HF 1 of the Ethereum 2.0 protocol. It builds on the [previous document for the behavior of an "honest validator" from Phase 0](../phase0/validator.md) of the Ethereum 2.0 protocol. This previous document is referred to below as the "Phase 0 document".
+
+HF 1 introduces a new type of committee: the sync committee. Sync committees are responsible for signing each block of the canonical chain and there exists an efficient algorithm for light clients to sync the chain using the output of the sync committees. See the [sync protocol](./sync-protocol.md) for further details on the light client sync. Under this network upgrade, validators will need to track their participation in this new committee type and produce the relevant signatures as required. Block proposers will need to incorporate the (aggregated) sync committee signatures into each block they produce.
+
+## Prerequisites
+
+All terminology, constants, functions, and protocol mechanics defined in the [HF 1 -- The Beacon Chain](./beacon-chain.md) doc are requisite for this document and used throughout. Please see this document before continuing and use as a reference throughout.
+
+## Custom types
+
+| Name | SSZ equivalent | Description |
+| - | - | - |
+| `SyncCommitteeIndex` | `uint64` | Index of a validator within a sync committee |
+
+## Containers
+
+### SyncCommitteeContribution
+
+```python
+class SyncCommitteeContribution(Container):
+    # Slot during which this contribution is broadcast
+    slot: Slot
+    # Position of the validator within the sync committee
+    index: SyncCommitteeIndex
```

Comment:
Pros of defining new type -- `SyncCommitteeIndex` -- instead of just using a `ValidatorIndex`. Presumable if you can verify this signature, you have map of validators <-> sync_committee_shuffling so both should be approximately equivalent

With `SyncCommitteIndex`, you do a lookup with the shuffling to get the validator index and verify the signature
With `ValidatorIndex`, you can verify signature direct, and need to make sure the validator index is contained within the current shuffling

Reply:
Ah, I see. The idea here is that if a validator is duplicated in the shuffling they would send two messages.

I'm not sure it's worth the new type here and duplicate message on network. The client can resolve a duplicate assignment by including the signature twice in the aggregate.

That said, I see *why* this is here now. open to discuss more